### ISSUE TITLE
Intersection type support

### ISSUE THREAD

**Issue by benjamind** [runem/lit-analyzer#18]
First off, I'm very excited by this plugin, excellent work!

I've ran into a small issue relating to union types.

```typescript
interface ComplexType {
	bool: Boolean;
}
type UnionType = ComplexType & { foo: string };
const complex: ComplexType = { bool: true };
const union: UnionType = { foo: "bah", bool: false };
const unionFunc = () => {
	return union;
};

// create a template which uses these properties
const template = html`<my-foo .complexProperty=${complex} .unionProperty=${union}></my-foo>`;
```
This errors on the `unionProperty` assignment with `Type 'UnionType' is not assignable to 'UnionType' lit-plugin(2322)` which is a thoroughly confusing error :-)

The complex type seems to validate correctly. Am I missing something?

If I pass the result of the `unionFunc` as `.unionProperty=${unionFunc()}` I still get the same unxpected error.

Interestingly if I pass the `unionFunc` as `.unionProperty=${unionFunc}` then I get no error, which is not what I'd expect since this is a function being assigned to a property, although this may be related to #3.

**Comment by runem** [comment]
Hi, thanks for creating this issue! I'm very happy that you like my plugin!

First of all I believe that you are thinking about [intersection types](https://www.typescriptlang.org/docs/handbook/advanced-types.html#intersection-types) and not union types. 

I have now created an issue on my repo [ts-simple-type](https://github.com/runem/ts-simple-type) tracking support for intersection types: https://github.com/runem/ts-simple-type/issues/4. In addition I'm almost finished implementing the support so you can expect a fix to come out within a couple of days :tada: 

You are correct that function types are not checked at the moment and that it's related to #3 . When that issue is closed `.unionProperty=${unionFunc}` should work as expected.

**Comment by benjamind** [comment]
Yes I meant intersection types (for some reason I always get that mixed up). 

Great to see this coming along so fast. This is going to be a very useful addition to developer experience for `lit-element` based components.

**Comment by benjamind** [comment]
I saw that [this issue](https://github.com/runem/ts-simple-type/issues/4) is now resolved on`ts-simple-type`. Does this mean we just need to upgrade this plugin to fix this issue?

**Comment by runem** [comment]
This fix is now live in version [1.1.6](https://marketplace.visualstudio.com/items?itemName=runem.lit-plugin) :-) I'm so sorry for the wait!
